npm/pkg/dataplane/policies/policymanager_linux.go

```diff
@@ -41,10 +42,92 @@ Known errors that we should retry on:
     Another app is currently holding the xtables lock. Stopped waiting after 60s.
 */
 
+/*
+Remove all NetworkPolicies that had a RemovePolicy() call.
+Add all NetworkPolicies which had an AddPolicy() call as the most recent call.
+
+Process removals first (e.g. in case a NetPol is updated)
+
+Examples:
+1. NetPol is updated. First there will be RemovePolicy() and then AddPolicy(). We will remove the Policy, then add it.
+2. NetPol is added, then removed before we reconcile. We will NOT add the Policy. We will try removing the Policy, but it will not be there.
+
+TODO: handle #2 above. Make sure we don't error out when failing to delete a NetPol that was never in the kernel
+*/
+func (pMgr *PolicyManager) reconcileDirtyNetPols() error {
+	pMgr.policyMap.Lock()
+	defer pMgr.policyMap.Unlock()
+
+	if len(pMgr.policyMap.linuxDirtyCache) == 0 {
+		return nil
+	}
+
+	toRemove := make([]*NPMNetworkPolicy, 0)
+	toAdd := make([]*NPMNetworkPolicy, 0)
+	for key, ops := range pMgr.policyMap.linuxDirtyCache {
+		policy, ok := pMgr.policyMap.cache[key]
+		if !ok {
+			metrics.SendErrorLogAndMetric(util.IptmID, "error: failed to find dirty policy in cache. key: %s", key)
+			continue
+		}
+
+		for _, op := range ops {
+			if op == remove {
+				toRemove = append(toRemove, policy)
+				break
+			}
+		}
+
+		if ops[len(ops)-1] == add {
+			toAdd = append(toAdd, policy)
+		}
+	}
+
+	if len(toRemove) > 0 {
+		if err := pMgr.removeAllPolicies(toRemove); err != nil {
```

Comment:
need to check if there any race conditions here ? 

1. If a policy is added and delete again within a same batch call. -> i think we are safe since the key wont be present in ou local cache
2. what if multiple updates come for a same policy -> safe for that too !? 

any other senarios ?

Reply:
TODO: UTs for all these edge cases

Reply:
irrelevant. moved to a new design